source/shared/components/App.js

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+class App extends Component {
+  addReader (book) {
+    console.log('test');
+    console.log(book);
+    // this.props.dispatch({
+    //   type: 'ADD_COUNT',
+    //   book,
+    // });
+  }
+
+  render () {
+    return <button onClick={this.addReader.bind(this)}>Test</button>;
```

Comment:
I take it this is what you're having trouble with? Generally speaking, you don't need `.bind()` in ES6. Try this?

```js
return <button onClick={ () => this.addReader(book) }>Test</button>;
```

But going further, you probably don't even need the `.addReader()` method:

```js
render ({dispatch}) {
  return <button onClick={ () => dispatch({ type: 'ADD_COUNT', book }); }>Test</button>;
}
```